scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/InProcessTaskExecutor.java

```diff
@@ -209,10 +214,26 @@ private Serializable execute(TaskContext taskContext, PrintStream output, PrintS
         if (taskContext.getPreScript() != null) {
             Script<?> script = taskContext.getPreScript();
             forkedTaskVariablesManager.replaceScriptParameters(script, thirdPartyCredentials, variables, error);
-            ScriptResult preScriptResult = scriptHandler.handle(script, output, error);
-            if (preScriptResult.errorOccured()) {
-                throw new TaskException("Failed to execute pre script: " + preScriptResult.getException().getMessage(),
-                                        preScriptResult.getException());
+            Map<String, String> genericInfo = taskContext.getInitializer().getGenericInformation();
+            if (genericInfo != null && genericInfo.containsKey("PRE_SCRIPT_AS_FILE")) {
+                String path = genericInfo.get("PRE_SCRIPT_AS_FILE");
+                ScriptEngineFactory factory = new ScriptEngineManager().getEngineByName(script.getEngineName())
+                                                                       .getFactory();
+                String extension = factory.getExtensions().get(0);
+                String code = script.fetchScript().trim();
+                try (FileWriter fw = new FileWriter(new File(path + "." + extension))) {
+                    fw.write(code);
+                    fw.close();
+                } catch (IOException e) {
+                    e.printStackTrace();
```

Comment:
exception should be forwarded as TaskException.
Never use e.printStackTrace() here